server/service/integration_enterprise_test.go

```diff
@@ -17636,67 +17636,89 @@ func (s *integrationEnterpriseTestSuite) TestBatchSoftwareCategories() {
 	srv := httptest.NewServer(handler)
 	t.Cleanup(srv.Close)
 
+	maintained1, err := s.ds.UpsertMaintainedApp(ctx, &fleet.MaintainedApp{
+		Name:             "1Password",
+		Slug:             "1password/darwin",
+		Platform:         "darwin",
+		UniqueIdentifier: "com.1password.1password",
+	})
+	require.NoError(t, err)
+
 	// do a request with an invalid category
-	rubyURL := srv.URL + "/ruby.deb"
+	rubyURL := srv.URL + "/dummy_installer.pkg"
 	softwareToInstall := []*fleet.SoftwareInstallerPayload{
-		{URL: rubyURL, SHA256: "foobar", Categories: []string{"Not Found"}, SelfService: true},
+		{URL: rubyURL, Categories: []string{"Not Found"}, SelfService: true},
+		{Slug: &maintained1.Slug, SelfService: true},
 	}
 	var batchResponse batchSetSoftwareInstallersResponse
 	s.DoJSON("POST", "/api/latest/fleet/software/batch", batchSetSoftwareInstallersRequest{Software: softwareToInstall}, http.StatusAccepted, &batchResponse, "team_name", team1.Name)
 	errMsg := waitBatchSetSoftwareInstallersFailed(t, s, team1.Name, batchResponse.RequestUUID)
 	require.Equal(t, "some or all of the categories provided don't exist", errMsg)
 
 	testCases := []struct {
-		desc       string
-		categories []string
+		desc                 string
+		categories           []string
+		fmaDefaultCategories []string
 	}{
 		{
 			desc:       "valid categories 1",
 			categories: []string{"Developer tools", "Browsers"},
 		},
 		{
 			desc:       "valid categories 2",
-			categories: []string{"Developer tools", "Browsers"},
+			categories: []string{"Communication", "Productivity"},
 		},
 		{
-			desc: "empty categories",
+			desc:                 "empty categories",
+			fmaDefaultCategories: []string{"Productivity"},
 		},
 	}
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			softwareToInstall[0].URL = rubyURL
-			softwareToInstall[0].SHA256 = ""
 			softwareToInstall[0].Categories = tc.categories
+			softwareToInstall[1].Categories = tc.categories
 			s.DoJSON("POST", "/api/latest/fleet/software/batch", batchSetSoftwareInstallersRequest{Software: softwareToInstall}, http.StatusAccepted, &batchResponse, "team_name", team1.Name)
 			packages := waitBatchSetSoftwareInstallersCompleted(t, s, team1.Name, batchResponse.RequestUUID)
-			require.Len(t, packages, 1)
```

Comment:
I removed a lot of these checks because they aren't super relevant to this test and are tested elsewhere already.